### ISSUE TITLE
Slow builds when there are many warnings/errors

### ISSUE THREAD

**Issue by lojjic** [webpack-contrib/uglifyjs-webpack-plugin#286]
This issue is to continue discussion from: https://github.com/webpack-contrib/uglifyjs-webpack-plugin/commit/0b11134fea8e92e92ec1b4b9bfe3cdcc9d8e640b#r28859767

Version 1.2.5 increased our build time by 4x, and I tracked it down to the above commit which implemented a sourcemap cache in a way that cache hits are impossible. Fixing the cache misses brings our build time back to normal, however there is some question whether this fix should be done with a `WeakMap` to allow GC or just use a simple `Map`.

**Comment by lojjic** [comment]
Another possible option (if I'm understanding the code flow correctly): remove the cache object altogether, go back to creating the `sourceMap = new SourceMapConsumer(...)` outside the loop, but make it so that it's created lazily only after we know there are errors or warnings in the result. Seems like this would meet the goals as I understand them: it's only created once per file, it's only created if there are actually errors or warnings, and it can be GC'd as soon as its variable pointer is out of scope.

**Comment by evilebottnawi** [comment]
@lojjic PR welcome, let's discussion on code
